plugins/org.yakindu.sct.doc.user/src/user-guide/c-domain.textile

```diff
@@ -255,6 +252,138 @@ p. With this definition we can use expressions like @t.a.x@, see the image below
+* retrieve the pointer pointing to a given variable,
+* pass pointers as parameters to functions, and
+* receive a pointer as a functions return value.
+
+h3(#cdom_declaring-pointer-variables). Declaring pointer variables
+
+Pointer variables are declared in a statechart's definition section as shown in the following example:
+
+bc. var n: int32_t
+var pInt: pointer<int32_t>
+var ppInt: pointer<pointer<int32_t> >
+var pTriangle: pointer<Triangle>;
+
+
+The declarations above declare
+* _n_ as a (non-pointer) variable of type _int32_t_,
+* _pInt_ as a pointer variable that is pointing to a variable of type _int32_t_,
+* _ppInt_ as a pointer that is pointing to a pointer that is pointing to a variable of type _int32_t_, and
+* _pTriangle_ as a pointer to a variable of the self-defined type _Triangle_.
+
+bq.. *Please note:*
+
+When closing the type specification in a pointer declaration with angle brackets, e.&nbsp;g. @pointer<pointer<int32_t> >@, the @>@ characters must be separated from each other by one or more white space characters. Writing e.&nbsp;g. @pointer<pointer<int32_t>>@ would result in an error.
+
+
+h3(#cdom_using-pointer-variables). Using pointer variables
+
+In order to actually assign a pointer to a pointer variable, you have to get hold of that pointer. To retrieve the pointer to a variable _v_, use _v_'s extension function _pointer_. That is, for a variable _v_, the expression _v.pointer_ evaluates to a pointer to _v_. Each variable has the _pointer_ extension function.
+
+Example: Let's say the pointer variable _pInt_ (declared in the example above) should point to the variable _n_. The following assignment accomplishes this:
+
+bc. pInt = n.pointer
+
+Similarly, a pointer to a pointer to a base type can be retrieved as follows:
+
+bc. ppInt = pInt.pointer;
+
+Or even:
+
+bc. ppInt = n.pointer.pointer
+
+In order to deference a pointer, i.&nbsp;e. to retrieve the value of what the pointer is pointing to, use the _value_ extension function, which is available on all pointer-type variables.
+
+Example: Let's say the pointer variable _pInt_ (declared in the example above) is pointing to some _int32_t_ variable. The value of the variable _pInt_ is pointing to should be assigned to the _int32_t_ variable _n_. The following assignment accomplishes this:
+
+bc. n = pInt.value;
+
+Similarly, if _ppInt_ points to a pointer pointing to some _int32_t_ variable, the following statement retrieves the latter's value:
+
+bc. n = ppInt.value.value;
+
+Passing pointer parameters to C functions is straightforward. Let's say you have a C function to rotate a triangle around a center point by a given angle. The C function is defined like this:
+
+bc. Triangle* rotateTriangle(Triangle* triangle, Point* centerPoint, float angle) { … }
+
+Provided the function is declared in an imported C header file, you can call it directly like this:
+
+bc. pTriangle2 = rotateTriangle(pTriangle, pCenterPoint, 45.0);
```

Comment:
I am not sure if I understand your question, @BeckmaR. The sample function receives a pointer to a _Triangle_ as one of its parameters and returns a pointer to the rotated _Triangle_. Only pointer here, so what's your point? If you wanted to suggest that the function could rotate the triangle in place and return _void_, yes, that would have been an option. However, I justed wanted to show how to pass a pointer to a function and to receive one as a result, and that's it.